docs/proposals/34-observability2.0-opentelemtry-operator-and-collectors.md

```diff
@@ -0,0 +1,156 @@
+
+- **Objective:** Ensure a smooth transition by interfacing with the current observability stack.
+- **Details:**
+  - **Fluent-bit Integration:** Configure Fluent-bit to forward logs to the collector’s `loki` receiver, maintaining compatibility with the Vali-based setup from *GEP-19*.
+  - **Vali Integration:** Use the `loki` exporter to output the shoot logs to the existing Vali backend.
+  - **Valitail Replacement:** Replace the current `valitail` system service on shoot nodes with the OpenTelemetry collector, ensuring it collects the same types of logs as before. This will be done by deploying a systemd service for the collector on each shoot node, in the same way as the current `valitail` service.
+  - **Event Logger Replacement:** In a later stage the current `event logger` shall be replaced with the `k8s-event` receiver in  the otel-collector shoot instance
+
+![Figure 2](assets/gep-34-shoot-observability-architecture.png)
+
+Example Systemd unit for the OpenTelemetry Collector in the shoot node:
+
+```ini
+[Unit]
+Description=opentelemetry-collector daemon
+Documentation=https://github.com/open-telemetry/opentelemetry-collector
+[Install]
+WantedBy=multi-user.target
+[Service]
+CPUAccounting=yes
+MemoryAccounting=yes
+CPUQuota=3%
+CPUQuotaPeriodSec=1000ms
+MemoryMin=29M
+MemoryHigh=400M
+MemoryMax=800M
+MemorySwapMax=0
+Restart=always
+RestartSec=5
+EnvironmentFile=/etc/environment
+ExecStartPre=/bin/sh -c "systemctl set-environment HOSTNAME=$(hostname | tr [:upper:] [:lower:])"
+ExecStart=/opt/bin/opentelemetry-collector --config=/var/lib/opentelemetry-collector/config/config
+```
+
+Example configuration for the OpenTelemetry Collector in the shoot node can be found [here](assets/gep-34-shoot-collector-config.yaml). For an overview of how it works, you can view the official [docs](https://opentelemetry.io/docs/collector/configuration/).
+
+A visualisation of the example config (Image generated with [otelbin](https://github.com/dash0hq/otelbin)):
+![Figure 3](assets/gep-34-collector-image.png)
+
+---
+
+## Alternatives
+
+### Direct Collector Deployment Without an Operator
+
+- **Description:** Deploy OpenTelemetry collectors in each shoot reconcile directly without using an operator.
+- **Pros:** Simplifies initial setup, avoids operator overhead.
+- **Cons:** Lacks scalability and management features (e.g., automatic upgrades, CR-based configuration), which are critical in Gardener’s multi-tenant environment.
+- **Reason Rejected:** The operator aligns with *GEP-19*’s precedent of using operators for observability components, ensuring consistency and scalability.
+
+---
+
+This GEP establishes a critical milestone in Gardener’s journey toward *Observability 2.0*. By deploying the OpenTelemetry operator and collectors, Gardener gains a standardized, extensible observability framework that integrates with its existing stack while paving the way for future enhancements like tracing and OTLP-native backends.
```

Comment:
While this GEP certainly is a stepping stone towards Observability 2.0, I would appreciate it if you would mention more explicitly what this is all about. In this GEP there is almost no mention of any signals but logging. This is fine, but then it should also be clear from reading the summary/motivation that this only handles logging.

Please adapt the document accordingly.

Reply:
Thanks, the summary is updated as requested.